test/e2e/tests/zone_aware_routing.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build e2e
+
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, ZoneAwareRoutingTest)
+}
+
+var ZoneAwareRoutingTest = suite.ConformanceTest{
+	ShortName:   "ZoneAwareRouting",
+	Description: "Resource with Zone Aware Routing enabled",
+	Manifests:   []string{"testdata/zone-aware-routing.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		t.Run("only local zone should get requests", func(t *testing.T) {
+			const sendRequests = 50
+
+			ns := "gateway-conformance-infra"
+			zoneAwareRoute := types.NamespacedName{Name: "zone-aware-http-route", Namespace: ns}
+			gwNN := types.NamespacedName{Name: "zone-aware-gtw", Namespace: ns}
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig,
+				suite.ControllerName,
+				kubernetes.NewGatewayRef(gwNN), zoneAwareRoute)
+
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/",
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+
+			// The EnvoyProxy pods have a nodeSelector targeting zone "0" so all requests
+			// should be routed to upstreams there.
+			expected := map[string]int{
+				"0": sendRequests,
+				"1": 0,
```

Comment:
arent there multiple workers, do we need more entries here ?

Reply:
If I'm reading the github action workflow configuration right it's only the control-plane node (NUM_WORKERS is unset) and I've now adjusted that to 1.

Reply:
cool sgtm cc @zirain